### PR TITLE
Woo: Add selectors for product variations from API

### PR DESCRIPTION
**Pull request by coderkevin**
This adds a selector to get the variations for a single product after
it's been fetched from the API.

It also adjusts the edit selectors for variations to include the API
data as a baseline for the edits to overlay on top of.

To Test: `npm test`

### CODE REVIEW

**Comment by coderkevin** [comment]
Rebased.

**Comment by coderkevin** [comment]
Moved selectors to `state/sites/product-variations` to match reducer/actions.

File: client/extensions/woocommerce/app/products/product-header.js [hunk 1]
```diff
@@ -13,21 +13,51 @@ import ActionHeader from 'woocommerce/components/action-header';
...
+}
+
+function renderSaveButton( onSave, product, isBusy, translate ) {
 	const saveExists = 'undefined' !== typeof onSave;
 	const saveDisabled = false === onSave;
 
-	const saveButton = saveExists &&
+	const saveLabel = ( product && ! isObject( product.id )
+		? translate( 'Update' )
+		: translate( 'Save & Publish' )
```
Comment (a8ci18n) [inline client/extensions/woocommerce/app/products/product-header.js]: Hi! I've found a possible matching string that has already been translated **67** times:
`translate( 'Save &amp; Publish' )` *ES Score: 8*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/extensions/woocommerce/app/products/product-update.js [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+
+	onTrash = () => {
+		// TODO: Add action dispatch to trash this product.
+	}
+
+	onSave = () => {
+		const { product, translate } = this.props;
+
+		const successAction = successNotice(
+			translate( '%(product)s successfully updated.', {
```
Comment (a8ci18n) [inline client/extensions/woocommerce/app/products/product-update.js]: Hi! I've found a possible matching string that has already been translated **57** times:
`translate( '%1$s updated successfully.' )` *ES Score: 10*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/product-variations-selectors)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Pull request merged**